hikari/channels.py

```diff
@@ -812,3 +813,37 @@ class GuildVoiceChannel(GuildChannel):
 
     If this is `0`, then assume no limit.
     """
+
+
+@attr.s(eq=True, hash=False, init=False, slots=False)
+class UnrecognisedChannel(PartialChannel):
+    """Channels that haven't been implemented in the library.
+
+    The attributes on this object are undocumented and dynamic.
+
+    Example
+    -------
+        >>> entry_info.foobar.baz
+
+    !!! note
+        This model has no slots and will have arbitrary undocumented attributes
+        (in it's `__dict__` based on the received payload).
+        The only attributes that are always garantied to be there are `app`,
+        `id`, `name` and `type`.
+    """
+
+    def __init__(
+        self,
+        /,
+        *,
+        app: traits.RESTAware,
+        id: snowflakes.Snowflake,
+        name: typing.Optional[str],
+        type: typing.Union[ChannelType, int],
+        payload: typing.Mapping[str, str],
+    ) -> None:
```

Comment:
Do we really want to use this style here? Wouldn't this have memory implications for the cache

Reply:
In what way?

Reply:
Ignore the 2nd part but whole this style is used in the audit log entry as its a compromise between handling unimplemented cases while not throwing away all useful data as there isn't any base partial data so an unknown type would have no attributes otherwise do we really want to introduce that behaviour in channels? Channels are a more core part of the lib and applying this approach here seems kinda questionable as its no longer a case of otherwise there'd just be no data and it would technically make all new channel type impls breaking since previous behaviour for that type would've likely had different attribute types and names.

Reply:
I would like to provide this as a nicer alternative than having `KeyError`s being raised everywhere. This at least provides a semi-useful object you can use to make some generic checks. An example of this would be so you can detect the channel create on recently implemented channels without having to update the library

Reply:
Could just return a partial channel or a partial channel that's been extended to include an optional guild id attribute

Reply:
This is what it is doing. The reason why I don't want to implement an optional guild_id is because I would like to keep the attributes provided by PartialChannel

Reply:
Not quite, what you're doing is returning an object with an arbitrary amount of attributes, if you really want to return a the data it'd be better to just have a other data dict attribute but I don't really see the point of providing all the data when this only exists to handle the case of an unexpected channel type softly and is intended to be ignored not implemented around

Reply:
Also when i say a partial channel i mean literal PartialChannel plus also GuildChannel since need to handle that case as well